resources/service.go

```diff
@@ -16,23 +16,54 @@ package resources
 
 import (
 	"errors"
+	"fmt"
 	"log"
 
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/labels"
+
+	"github.com/Mirantis/k8s-AppController/client"
 )
 
 type Service struct {
-	Service *api.Service
-	Client  unversioned.ServiceInterface
+	Service   *api.Service
+	Client    unversioned.ServiceInterface
+	APIClient client.Interface
 }
 
-func serviceStatus(s unversioned.ServiceInterface, name string) (string, error) {
-	_, err := s.Get(name)
+func serviceStatus(s unversioned.ServiceInterface, name string, apiClient client.Interface) (string, error) {
+	service, err := s.Get(name)
+
 	if err != nil {
 		return "error", err
 	}
 
+	log.Printf("Checking service status for selector %v", service.Spec.Selector)
+	for k, v := range service.Spec.Selector {
+		stringSelector := fmt.Sprintf("%s=%s", k, v)
+		log.Printf("Checking status for %s", stringSelector)
+		selector, err := labels.Parse(stringSelector)
+
+		options := api.ListOptions{LabelSelector: selector}
+
+		pods, err := apiClient.Pods().List(options)
+		if err != nil {
+			return "error", err
+		}
+		for _, pod := range pods.Items {
+			log.Printf("Checking status for pod %s", pod.Name)
+			p := NewPod(&pod, apiClient.Pods())
```

Comment:
This looks like a bit of redundant (NewPod). Mb it worth exporting podStatus from the module rather than creating extra level of indirection?